crates/wasip3/src/http_ext/conversions.rs

```diff
@@ -0,0 +1,102 @@
+use super::to_internal_error_code;
+use crate::http::types::{ErrorCode, Fields, HeaderError, Headers, Method, Scheme};
+use std::convert::TryFrom;
+
+impl TryFrom<Scheme> for http::uri::Scheme {
+    type Error = ErrorCode;
```

Comment:
While this makes sense for the way these are used by other code in this crate I think it would be more generally useful for these fallible conversions to return their "native" error types (e.g. `http::uri::InvalidUri` here) and convert the errors at call sites as needed.

Reply:
Updated, PTAL